Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/cache/delta_fifo.go

```diff
@@ -154,10 +156,30 @@ func (f *DeltaFIFO) Update(obj interface{}) error {
 	return f.queueActionLocked(Updated, obj)
 }
 
-// Delete is just like Add, but makes an Deleted Delta.
+// Delete is just like Add, but makes an Deleted Delta. If the item does not
+// already exist, it will be ignored. (It may have already been deleted by a
+// Replace (re-list), for example.
 func (f *DeltaFIFO) Delete(obj interface{}) error {
+	id, err := f.KeyOf(obj)
+	if err != nil {
+		return KeyError{obj, err}
+	}
 	f.lock.Lock()
 	defer f.lock.Unlock()
+	if f.knownObjects == nil {
+		if _, exists := f.items[id]; !exists {
+			// Presumably, this was deleted when a relist happened.
+			// Don't provide a second report of the same deletion.
+			return nil
```

Comment:
@deads2k @liggitt 
maybe we're using it wrong, but the result of this change is that we never get deleted events because we don't provide a list of knownobjects, and the item is never already in the queue.  Seems like this should still be adding a Deleted event to the queue with a DeletedFinalStateUnknown.  If i comment this line out (the return nil) the tests pass again.